### ISSUE TITLE
Strict mode problem : 'with'

### ISSUE THREAD

**Issue by sixrandanes** [basti1302/traverson-angular#20]
Hello,

After including the traverson-angular lib in my project, I am facing an issue when I run my build.
In my build, I have a ng-annotate task and that task just failed :

```
error: couldn't process source due to parse error
'with' in strict mode
```

Without this lib, everythings works fine.
The problem seems to be from that line with keyword "with" 

```
          var isNode = "undefined" != typeof module && !!module.exports,
              vm = isNode ? _dereq_("vm") : {runInNewContext: function(expr, context) {
                  with (context) return eval(expr);
                }};
```
I am running my project with the last version of node 5.3.0.

Any help would be appreciated.

Thanks,

**Comment by sixrandanes** [comment]
I have found a workaround by downgrading the version of ng-annotate (new release 2 days ago).
My build is now running properly.

But still thinking the keyword with should be refactored in the code above.
What do you think ?

**Comment by basti1302** [comment]
I had a quick look, this seems to come from the JSONPath dependency of Traverson. Good news, this has already been [changed](https://github.com/s3u/JSONPath/pull/28) in JSONPath.

Chances are this will go away when I update the JSONPath version and build new releases. 

However, I wonder why in your build ng-annotate processes a third party lib like Traverson. You might want to restrict ng-annotate to only process your own sources.

**Comment by basti1302** [comment]
The "fix" has landed in release 3.1.1.

**Comment by sixrandanes** [comment]
Yes you're right for the process ng-annotate. I had to change it in order to take into account only our sources.

Thanks a lot for your help.
